app/forms/candidate_interface/training_locations_form.rb

```diff
@@ -0,0 +1,37 @@
+module CandidateInterface
+  class TrainingLocationsForm
+    include ActiveModel::Model
+    include Rails.application.routes.url_helpers
+
+    LOCATION_OPTIONS = %w[anywhere specific].freeze
+
+    attr_accessor :preference, :training_locations
+    validates :training_locations, inclusion: { in: LOCATION_OPTIONS }
+
+    def self.build_from_preference(preference)
+      new({ training_locations: preference.training_locations }, preference:)
+    end
+
+    def initialize(attributes = {}, preference:)
+      @preference = preference
+      super(attributes)
+    end
+
+    def save!
+      ActiveRecord::Base.transaction do
+        @preference.update!(training_locations:)
+        if @preference.reload.training_locations_specific? && @preference.location_preferences.empty?
+          LocationPreferences.add_default_location_preferences(preference: @preference)
```

Comment:
We pre-populate the location preferences for the next step if they are empty. If they have already added some, we don't override them.